tests/e2e_tests/conftest.py

```
@@ -22,6 +23,21 @@ def pytest_runtest_makereport(item, call):
+def _dump_container_logs_on_failure(request, docker_containers):
+    """On test failure, write full docker logs to tests/e2e_tests/logs/<test>/<container>.log."""
+    rep = getattr(request.node, "rep_call", None)
+    if rep is None or not rep.failed:
+        return
+    log_dir = Path(__file__).parent / "logs" / f"{request.node.name}_{int(time.time())}"
+    log_dir.mkdir(parents=True, exist_ok=True)
+    for c in docker_containers:
+        try:
+            c.reload()
+            (log_dir / f"{c.name}.log").write_bytes(c.logs())
+        except Exception as e:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)